dash/testing/application_runners.py

```diff
@@ -248,28 +249,39 @@ def __init__(self, keep_open=False, stop_timeout=3):
         self.proc = None
 
     # pylint: disable=arguments-differ
-    def start(self, app, start_timeout=2):
-        """Start the server with waitress-serve in process flavor."""
+    def start(self, app, start_timeout=2, cwd=None):
+        """Start the server with subprocess and Rscript."""
 
         # app is a R string chunk
-        if not (os.path.isfile(app) and os.path.exists(app)):
+        if (os.path.isfile(app) and os.path.exists(app)):
+            # app is already a file in a dir - use that as cwd
+            if not cwd:
+                cwd = os.path.dirname(app)
+        else:
             path = (
                 "/tmp/app_{}.R".format(uuid.uuid4().hex)
                 if not self.is_windows
                 else os.path.join(
                     (os.getenv("TEMP"), "app_{}.R".format(uuid.uuid4().hex))
                 )
             )
-            logger.info("RRuner start => app is R code chunk")
-            logger.info("make a temporay R file for execution=> %s", path)
-            logger.debug("the content of dashR app")
+            logger.info("RRunner start => app is R code chunk")
+            logger.info("make a temporary R file for execution => %s", path)
+            logger.debug("content of the dashR app")
             logger.debug("%s", app)
 
             with open(path, "w") as fp:
                 fp.write(app)
 
             app = path
 
+            # try to find the path to the calling script to use as cwd
+            if not cwd:
+                for entry in inspect.stack():
+                    if "/dash/testing/" not in entry[1].replace("\\", "/"):
```

Comment:
a little bit scared about this hack here

Reply:
Yes, it's ugly. Do you see an alternative? I suppose if we're too worried about it we could document the official usage pattern as "start your server with `dashr.run_server(app, cwd=os.path.dirname(__file__))`" but it would be nice if we can avoid that.

Reply:
can we just use the `os.path.dirname(__file__)` by default (in case `cwd=None`) ?

Reply:
but we don't want `__file__` from here, we want it from the user's test module